pandas/tests/test_reshape.py

```diff
@@ -411,6 +411,96 @@ def test_dataframe_dummies_with_categorical(self):
                              ]]
         assert_frame_equal(result, expected)
 
+    # GH12402 Add a new parameter `drop_first` to avoid collinearity
+    def test_basic_drop_first(self):
+        # Basic case
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_series_index = Series(s_list, list('ABC'))
+
+        expected = DataFrame({'b': {0: 0.0,
+                                    1: 1.0,
+                                    2: 0.0},
+                              'c': {0: 0.0,
+                                    1: 0.0,
+                                    2: 1.0}})
+
+        result = get_dummies(s_list, sparse=self.sparse, drop_first=True)
+        assert_frame_equal(result, expected)
+
+        result = get_dummies(s_series, sparse=self.sparse, drop_first=True)
+        assert_frame_equal(result, expected)
+
+        expected.index = list('ABC')
+        result = get_dummies(s_series_index, sparse=self.sparse, drop_first=True)
+        assert_frame_equal(result, expected)
+
+    # Test the case that categorical variable only has one level.
+    def test_basic_drop_first_one_level(self):
+        result = get_dummies(list('aaa'), sparse=self.sparse, drop_first=True)
+        self.assertEqual(result.empty, True)
```

Comment:
compare this with an actual constructed empty frame as this will verify that the indexes are correct.